core/block_validator.go

```diff
@@ -68,15 +68,25 @@ func (v *BlockValidator) ValidateBody(block *types.Block) error {
 	if hash := types.DeriveSha(block.Transactions(), trie.NewStackTrie(nil)); hash != header.TxHash {
 		return fmt.Errorf("transaction root hash mismatch (header value %x, calculated %x)", header.TxHash, hash)
 	}
-	// Withdrawals are present after the Shanghai fork.
-	if header.WithdrawalsHash != nil {
+	if v.config.IsShanghai(header.Time) {
```

Comment:
I looked at this again, and I'd be wary of adding this check in this specific function, as it's not only used during boost relay block validation